{
  "comments": [
    {
      "unresolved": true,
      "key": {
        "uuid": "12f862f4_7858154a",
        "filename": "src/libANGLE/renderer/vulkan/SyncVk.cpp",
        "patchSetId": 4
      },
      "lineNbr": 205,
      "author": {
        "id": 1340649
      },
      "writtenOn": "2023-10-04T00:29:20Z",
      "side": 1,
      "message": "have seen it done both ways. is there a coding guideline w.r.t lambda captures?\noutside lambda scope no one uses `renderer` we can just capture it there right? -\n`[renderer \u003d context-\u003egetRenderer(), context...]`",
      "range": {
        "startLine": 205,
        "startChar": 31,
        "endLine": 205,
        "endChar": 39
      },
      "revId": "4c30b216b6041dab855e4cdfd931bf4e0c167dc3",
      "serverId": "3ce6091f-6c88-37e8-8c75-72f92ae8dfba"
    },
    {
      "unresolved": false,
      "key": {
        "uuid": "65dee844_78f3a787",
        "filename": "src/libANGLE/renderer/vulkan/SyncVk.cpp",
        "patchSetId": 4
      },
      "lineNbr": 205,
      "author": {
        "id": 1300114
      },
      "writtenOn": "2023-10-04T02:01:59Z",
      "side": 1,
      "message": "No guideline, and yes we could capture there. Pretty harmless either way. Having a line of `RendererVk *renderer \u003d ...` at the top of the function or something like that is fairly common in the code at this point.",
      "parentUuid": "12f862f4_7858154a",
      "range": {
        "startLine": 205,
        "startChar": 31,
        "endLine": 205,
        "endChar": 39
      },
      "revId": "4c30b216b6041dab855e4cdfd931bf4e0c167dc3",
      "serverId": "3ce6091f-6c88-37e8-8c75-72f92ae8dfba"
    },
    {
      "unresolved": true,
      "key": {
        "uuid": "140e9870_f2e790c4",
        "filename": "src/libANGLE/renderer/vulkan/SyncVk.cpp",
        "patchSetId": 4
      },
      "lineNbr": 207,
      "author": {
        "id": 1340649
      },
      "writtenOn": "2023-10-04T13:52:17Z",
      "side": 1,
      "message": "nit - should add something here to indicate that it is an unlocked wait",
      "range": {
        "startLine": 207,
        "startChar": 8,
        "endLine": 207,
        "endChar": 26
      },
      "revId": "4c30b216b6041dab855e4cdfd931bf4e0c167dc3",
      "serverId": "3ce6091f-6c88-37e8-8c75-72f92ae8dfba"
    },
    {
      "unresolved": true,
      "key": {
        "uuid": "88b96563_0c9e7d2c",
        "filename": "src/libANGLE/renderer/vulkan/SyncVk.cpp",
        "patchSetId": 4
      },
      "lineNbr": 490,
      "author": {
        "id": 1340649
      },
      "writtenOn": "2023-10-04T00:29:20Z",
      "side": 1,
      "message": "how do we handle cases where `mExternalFence` gets deleted while we are CPU waiting outside the shared mutex lock?",
      "range": {
        "startLine": 490,
        "startChar": 71,
        "endLine": 490,
        "endChar": 85
      },
      "revId": "4c30b216b6041dab855e4cdfd931bf4e0c167dc3",
      "serverId": "3ce6091f-6c88-37e8-8c75-72f92ae8dfba"
    },
    {
      "unresolved": true,
      "key": {
        "uuid": "7efbffff_10202af8",
        "filename": "src/libANGLE/renderer/vulkan/SyncVk.cpp",
        "patchSetId": 4
      },
      "lineNbr": 490,
      "author": {
        "id": 1300114
      },
      "writtenOn": "2023-10-04T09:30:40Z",
      "side": 1,
      "message": "It\u0027s a `shared_ptr` being copied into the closure of this lambda, can that happen?\n\nNot very familiar with the external fence stuff, can it get cleaned up in some way before the shared_ptr ref count gets to 0?",
      "parentUuid": "88b96563_0c9e7d2c",
      "range": {
        "startLine": 490,
        "startChar": 71,
        "endLine": 490,
        "endChar": 85
      },
      "revId": "4c30b216b6041dab855e4cdfd931bf4e0c167dc3",
      "serverId": "3ce6091f-6c88-37e8-8c75-72f92ae8dfba"
    },
    {
      "unresolved": false,
      "key": {
        "uuid": "2ba66f96_d2eee404",
        "filename": "src/libANGLE/renderer/vulkan/SyncVk.cpp",
        "patchSetId": 4
      },
      "lineNbr": 490,
      "author": {
        "id": 1340649
      },
      "writtenOn": "2023-10-04T13:52:17Z",
      "side": 1,
      "message": "\u003e It\u0027s a shared_ptr\n\nhmmm, i was thinking of the case of sync being destroyed while we CPU wait outside the lock but i missed the shared pointer bit. I suppose when `fence` goes out of scope, clean up will be handled appropriately. all good, thanks",
      "parentUuid": "7efbffff_10202af8",
      "range": {
        "startLine": 490,
        "startChar": 71,
        "endLine": 490,
        "endChar": 85
      },
      "revId": "4c30b216b6041dab855e4cdfd931bf4e0c167dc3",
      "serverId": "3ce6091f-6c88-37e8-8c75-72f92ae8dfba"
    },
    {
      "unresolved": true,
      "key": {
        "uuid": "7efb05c6_577c8013",
        "filename": "src/libANGLE/renderer/vulkan/SyncVk.cpp",
        "patchSetId": 4
      },
      "lineNbr": 492,
      "author": {
        "id": 1340649
      },
      "writtenOn": "2023-10-04T13:52:17Z",
      "side": 1,
      "message": "nit - should add something here to indicate that it is an unlocked wait",
      "range": {
        "startLine": 492,
        "startChar": 8,
        "endLine": 492,
        "endChar": 26
      },
      "revId": "4c30b216b6041dab855e4cdfd931bf4e0c167dc3",
      "serverId": "3ce6091f-6c88-37e8-8c75-72f92ae8dfba"
    }
  ]
}